jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -1,68 +1,146 @@
+
+        // Find the DisplayMessageFragment on restart.
+        display_message_fragment = (DisplayMessageFragment)
+                fragmentManager.findFragmentById(R.id.fragment_container);
+
+        if (isDisplayMessageFragmentVisible) {
+            // Set width of fragment layout container to fill half the screen.
+            changeFragmentLayout(findViewById(R.id.fragment_container),
+                    ViewGroup.LayoutParams.MATCH_PARENT);
+        }
+
+        // Create DisplayMessageFragment if it hasn't been created yet. Show by default.
+        if (display_message_fragment == null) {
+            FragmentTransaction ft = fragmentManager.beginTransaction();
+            display_message_fragment = new DisplayMessageFragment();
+            ft.add(R.id.fragment_container, display_message_fragment).commitNow();
+        }
+
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle savedInstanceState) {
+        // Save the state of the DisplayMessageFragment.
+        savedInstanceState.putBoolean(IS_MESSAGE_FRAGMENT_VISIBLE, isDisplayMessageFragmentVisible);
+        super.onSaveInstanceState(savedInstanceState);
     }
 
     /** Called when the user clicks the Send button */
     public void sendMessage(View view) {
 
-        // The message entered by the user.
-        String message = editText.getText().toString();
+        // If the display message fragment is visible show the message there.
+        if (display_message_fragment != null) {
+            display_message_fragment.updateMessage(main_fragment.getMessage());
+        }
 
-        // If orientation is in landscape add message to fragment.
-        if (getResources().getConfiguration().orientation ==
-                Configuration.ORIENTATION_LANDSCAPE) {
+    }
 
-            // Get the fragment that shows the message.
-            if (display_message_fragment == null) {
-                FragmentManager fragmentManager = getSupportFragmentManager();
-                display_message_fragment =
-                        fragmentManager.findFragmentById(R.id.displaymessage_fragment);
-            }
+    /** Called when the user clicks the show/hide button */
+    public void showHide(View view) {
 
-            // Edit the text view (show written message).
-            TextView textView = (TextView)
-                    display_message_fragment.getView().findViewById(R.id.display_message);
-            textView.setText(message);
+        // If the displaying fragment has not been initialized return.
+        if (display_message_fragment == null) {
+            return;
+        }
 
-        // If orientation is in portrait start new activity.
-        } else {
-            // Create intent.
-            Intent intent = new Intent(this, DisplayMessageActivity.class);
+        // Setup FragmentManager and FragmentTransaction.
+        View fragment_container = findViewById(R.id.fragment_container);
```

Comment:
Performance wise, would you say it's better to get the `fragment_container` when the `DisplayMessageFragment` is created in `onCreate` (even though you actually may never press the show/hide button) or make a check here to see if it has already been found? Then you have to do the check every time you press the button. I mean, it's a small matter, but I'm just curious :)